### PR TITLE
#259 Container registry credentials docs

### PR DESCRIPTION
**Pull request by llewellyn-sl**
Update Credentials page with User Credentials and Container registry credentials sections.
Update the following Compute Environment guides with additional credential creation steps and container registry credentials note:
-AWS Batch
-Azure Batch
-EKS
-Google Cloud
-GKE

### CODE REVIEW

**Comment by pditommaso** [comment]
Ok, nice. However, I think it would be better to keep Container credentials separated from the Compute Environment configuration. 

Users may want to configure it even without using the any compute env in tower and they will access it via Wave + nextflow

File: docs/credentials/overview.md [hunk 1]
```diff
@@ -6,10 +6,35 @@ description: 'Step-by-step instructions to set up credentials in Nextflow Tower.
...
+From version 22.4, Tower supports the configuration of credentials for container registry services. These credentials can be created from the Credentials tab using these steps:
+
+
+1. Select **Add Credentials**. 
+2. Enter a unique name in the **Name** field using alphanumeric characters, dashes, or underscores. 
+3. From the **Provider** drop-down list, select **Container registry**. The New Credentials form now displays additional fields to be completed: 
+
+![](_images/container_registry_credentials_blank.png)
+
+4. Enter the credentials to be used to access the container registry, starting with the **User name**.
```
Comment (abhi18av) [inline docs/credentials/overview.md:36]: Inline images are not the best ideas within the list as it triggers a recoun after the image.

That's the reason why we have used bold numbers to achieve the same effect previously.

image

**Review by abhi18av — CHANGES_REQUESTED** [review]
Looking good in general, added some suggestions for improvements :)

**Comment by gwright99** [comment]
> Okay. So is there no use case where users would want to use the container credentials in a Tower compute env? I just added a note to each of the compute env pages which links back to the Credentials section where the container creds instructions are.
>
>If this is entirely separate, the best approach is to add sections to Credentials as follows (since the container creds steps for each registry service are different):

We've had examples before where a client needed to connect their AWS Batch compute environment to a private JFrog registry. We initially had to place the instructions into a modified Launch Template. I believe they migrated to a container credential subsequently.

**Comment by llewellyn-sl** [comment]
>We've had examples before where a client needed to connect their AWS Batch compute environment to a private JFrog >registry. We initially had to place the instructions into a modified Launch Template. I believe they migrated to a container >credential subsequently.

Matteo and Abhi mentioned we had a successful JFrog integration. May be worthwhile adding this to its own page in the container credentials section. Currently changing the IA as follows in the Credentials menu:

>Credentials
>>User credentials (one-page overview with generic user credentials steps

>>Container credentials (one-page overview with links to registry-specific instructions for the below)

>>>Docker
>>>AWS
>>>Google
>>>Azure
>>>Quay

File: docs/compute-envs/google-cloud.md [hunk 2]
```diff
@@ -119,14 +119,20 @@ To create a new compute environment for Google Cloud in Tower:
...
 
 6. Enter the **Service account key** for your Google Cloud account. This key was created in the [previous section](#create-service-account-key).
 
     ![](_images/google_tower_credentials.png)
 
+    !!! tip "Multiple credentials"
+        You can create multiple credentials in your Tower environment.
+
+    !!! note "Container registry credentials"
+        From version 22.4, Tower supports the use of credentials for container registry services. These credentials can be created from the [Credentials](../credentials/overview.md/#container-registry-credentials) tab.
```
[the remark this answers is not in the document]
Comment (llewellyn-sl) [reply docs/compute-envs/google-cloud.md:134]: @pditommaso @swampie 22.3 or 22.4?

File: docs/credentials/overview.md (hunk 1 above)
Comment (llewellyn-sl) [reply docs/credentials/overview.md:36]: Thanks for catching this. Updated all pages with the in-line image to use numbering in bold. Preview is good.

File: docs/compute-envs/google-cloud.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (pditommaso) [reply docs/compute-envs/google-cloud.md:134]: 👉 22.3

**Comment by pditommaso** [comment]
> Okay. So is there no use case where users would want to use the container credentials in a Tower compute env? I just added a note to each of the compute env pages which links back to the Credentials section where the container creds instructions are

We actually support both. Tower credentials can be used by Tower CE via Wave or directly when using Nextlow (always via Wave).

This why I think they should have described separately

**Review by abhi18av — APPROVED** [review]

**Pull request merged**